engine.io-server/src/main/java/io/socket/engineio/server/EngineIoSocket.java

```diff
@@ -303,17 +309,15 @@ private void flush() {
     }
 
     private void resetPingTimeout() {
-        if(mPingTimeout != null) {
-            mPingTimeout.cancel();
+        if(mPingTimerScheduledReference != null) {
+            mPingTimerScheduledReference.cancel(false);
         }
 
-        mPingTimeout = new TimerTask() {
-            @Override
-            public void run() {
-                onClose("ping timeout", null);
-            }
-        };
-        mPingTimer.schedule(mPingTimeout,
-                mServer.getOptions().getPingInterval() + mServer.getOptions().getPingTimeout());
+        Runnable pingTimeoutTask = () -> onClose("ping timeout", null);
```

Comment:
This can probably be moved into a final member instance to reduce object creations.

Reply:
Yes this makes sense. Will do.